meerk40t/core/elements.py

```diff
@@ -3258,21 +3272,562 @@ def element_grid(
+            ),
+            input_type=("elements", None),
+            output_type="elements",
+        )
+        def element_polyshape(
+            corners,
+            x_pos,
+            y_pos,
+            radius,
+            startangle=None,
+            inscribed=None,
+            radius_inner=None,
+            alternate_seq=None,
+            data=None,
+            **kwargs,
+        ):
+            if corners is None:
+                raise SyntaxError
+            if corners < 3:
+                raise SyntaxError(_("A polyshape needs to have at least three corners"))
+
+            if x_pos is None:
+                raise SyntaxError(
+                    _(
+                        "Please provide at least one additional value (which will act as radius then)"
+                    )
+                )
+            else:
+                if not x_pos.is_valid_length:
+                    raise SyntaxError("x_pos: " + _("This is not a valid length"))
+
+            if y_pos is None:
+                y_pos = Length(0)
+            else:
+                if not y_pos.is_valid_length:
+                    raise SyntaxError("y_pos: " + _("This is not a valid length"))
+            # do we have something like 'polyshape 3 4cm' ? If yes, reassign the parameters
+            if radius is None:
+                radius = x_pos
+                x_pos = Length(0)
+                y_pos = Length(0)
+            else:
+                if not radius.is_valid_length:
+                    raise SyntaxError("radius: " + _("This is not a valid length"))
+
+            if startangle is None:
+                startangle = Angle.parse("0deg")
+            if inscribed is None:  # Use circumscribed circle by default
+                inscribed = False
+
+            if alternate_seq is None:
+                if radius_inner is None:
+                    alternate_seq = 0
+                else:
+                    alternate_seq = 1
+
+            if radius_inner is None:
+                radius_inner = radius
+            else:
+                if not radius_inner.is_valid_length:
+                    raise SyntaxError(
+                        "radius_inner: " + _("This is not a valid length")
+                    )
+
+            x_pos = x_pos.value(
+                ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+            )
+            y_pos = y_pos.value(
+                ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+            )
+            radius = radius.value(
+                ppi=1000, relative_length=bed_dim.bed_width * MILS_IN_MM
+            )
+            if (
+                isinstance(radius, Length)
+                or isinstance(x_pos, Length)
+                or isinstance(y_pos, Length)
+            ):
+                raise SyntaxError
+            if inscribed:
+                radius = radius / cos(pi / corners)
+
+            radius_inner = radius_inner.value(ppi=1000, relative_length=radius)
```

Comment:
Clever. Shows a firm understanding of the use of relative_length there for % values.